### PR TITLE
Adjust the icons and text of the binding connected blocks

### LINKED ISSUE #61515: Try using synced color on the block icon for overrides
Instead of using the "connected" icon to indicate a synced pattern, in place of the block icon, let's try using the synced color applied to the block icon instead. Maintaining the block icon is helpful for understanding block selection when using list view or content only view. And this reduces adding a new icon/concept to comprehend. 

Iteration: #59819

### Visual
CleanShot 2024-05-09 at 14 54 09

CleanShot 2024-05-09 at 16 02 47

### PR DESCRIPTION
**Pull request by kevin940726**
## What?

Close https://github.com/WordPress/gutenberg/issues/61515 and https://github.com/WordPress/gutenberg/issues/61547.

Adjust the icons and the text descriptions of the binding connected blocks (pattern overrides, connected post meta, etc.)

## Why?

The generic connection icon is not that helpful and the block type names should be helpful.

## How?

Adjust `<BlockBindingsToolbarIndicator>`.

## Testing Instructions

- Create some patterns with overrides and some post meta connected blocks.

You can paste the markup here to create an empty post-meta connected paragraph.
```html
<!-- wp:paragraph {"metadata":{"name":"connect","bindings":{"content":{"source":"core/post-meta","args":{}}}}} -->
<p>Connected</p>
<!-- /wp:paragraph -->
```

## Screenshots or screencast 
**A pattern overrides block**
image

**Multiple pattern overrides blocks**
image

**A post-meta connected block**
image

**Multiple post-meta connected blocks**
image

### CODE REVIEW

File: packages/block-editor/src/components/block-bindings-toolbar-indicator/index.js [hunk 1]
```diff
@@ -1,19 +1,135 @@
...
-				as={ 'div' }
-				aria-label={ _x( 'Connected', 'block toolbar button label' ) }
-				className="block-editor-block-bindings-toolbar-indicator"
-			>
-				<Icon icon={ connection } size={ 24 } />
+			<ToolbarItem>
+				{ ( toggleProps ) => (
+					<DropdownMenu
+						className="block-editor-block-bindings-toolbar-indicator"
+						label={ firstBlockTitle }
```
Comment (kevin940726) [inline packages/block-editor/src/components/block-bindings-toolbar-indicator/index.js:101]: The `aria-label` here doesn't match the visual label below. This is using the block title's name, but the visual is using the block's `metadata.name`. Not sure about the accessibility concern here and how we should fix it. @richtabor Any idea?

**Comment by richtabor** [comment]
I don't think we should use the override/connection name in the toolbar. It's enough to surface the details in the popover.

**Comment by richtabor** [comment]
I also think the copy in the proposed issue is a bit clearer and informative (with lower case block name as well).

**Comment by jameskoster** [comment]
> I don't think we should use the override/connection name in the toolbar

Agree with this, especially as the name could be long. However, by removing the label we'd see the same icon consecutively when editing the source which is a bit odd:

Screenshot 2024-05-14 at 11 22 02

Maybe that isn't supposed to be there though.

---

The 'help text' is useful, but shouldn't live in a `role="menu"` element. Could that just be a tooltip? 

> Paragraph connected to: 'x'

**Comment by jarekmorawski** [comment]
I wonder if the connection's status message should be shown as a tooltip instead. Currently, it's a button that doesn't trigger any action, which can confuse users (including those using a screen reader).

image

If we wanted to keep it as a popover, could we let users unlink or change the connection? On that note, can we make the messages more specific? It'd be helpful to mention what the block is connected to.

image


When multiple blocks are selected, we could list all connections. Clicking one would take the user to a specific

image

**Comment by talldan** [comment]
@richtabor @jameskoster @jarekmorawski Is there a conclusion about the design direction we want to go in?

Wanted to flag as well that the 6.6 beta is rapidly approaching (a week on Friday for Gutenberg), so might be good to think about it in terms of what's achievable within that timescale, and what we can iterate on for 6.7.

**Comment by jasmussen** [comment]
Thanks for the ping. I [just discussed](https://github.com/WordPress/gutenberg/pull/61811#issuecomment-2124244048) with @artemiomorales what we might have in time for next Friday, and IMO our opportunity here is to ship block bindings with some of the base UI that we _know_ we need, and then learn not only from beta feedback, but from what people do in the 6.6 phase, to build out any additionally needed affordances for 6.7. 

Especially some of the pieces that have been proposed for the block toolbar itself have received various bits of critique both across design and accessibility, which to me suggests that could be a piece to refine for the next release. Things like the use of purple, the dot indicator, and more. 

So what could a V1 look like? The block inspector is the most prominent piece, footprint wise, and the main interface for block connections:

Inspectors for Buttons and Image, with a Block Connections panel

It's unclear whether folks who use block bindings would close the inspector, given it's a slightly more advanced than usual flow, but if they did, the pre-publish flow would surface if they had edited bound blocks, and give an option to uncheck that:

Pre publish flow with connections and a checkbox surfaced

It would benefit the flow to surface connections in more states. When a URL is connected, show it as such in the URL popover. When an image URL is connected, show that in the image dropdown, and so on. It's less clear how far we can get with those in time for beta 1, but a minimal version could be this:

minimal version of button and image toolbars, with popovers showing indicators

That mockup intentionally omits colors and other affordances added already in trunk. That's not necessarily a suggestion to pull those out, I'd keep those at least for beta 1, but then in that period, learn, and refine, and then decide whether to keep those pieces, or refine them in 6.7.

What do you think?

**Comment by jameskoster** [comment]
Is the `🔌 Connected` badge beneath the block description required, isn't that effectively communicated via the Connections panel?

I wonder if the Connections panel could include the 🔌 icon somewhere, to conceptually connect (pun intended) that panel with the locations in which the icon appears. Perhaps in the title, or per-row like color swatches and other similar UIs?

In the save panel there's a 'Block bindings' item. Could this be 'Connections' to align with the panel (or vice versa)?

**Comment by richtabor** [comment]
> Is the 🔌 Connected badge beneath the block description required, isn't that effectively communicated via the Connections panel?

Had the same thought. Also, the unblock icon.

**Comment by jarekmorawski** [comment]
> I wonder if the Connections panel could include the 🔌 icon somewhere, to conceptually connect (pun intended) that panel with the locations in which the icon appears. Perhaps in the title, or per-row like color swatches and other similar UIs?

Agreed. Could something like this work? But with the correct icon, of course.

image

**Comment by jasmussen** [comment]
Yes, that could work. Feel free to tinker in [the Figma](https://www.figma.com/design/W2SO7yJG09O65q7JjfXZ4m/Block-connections?node-id=618%3A5643&t=xUUhKK3TkOpyDKMf-1).

**Comment by jarekmorawski** [comment]
I sketched a few more options, including one where the connection icon could be used as both a status indicator and a control. When clicked, it would detach the block (if allowed/enabled). This logically aligns with the `Alternative text` example from Joen's design.

image

**Comment by jameskoster** [comment]
Icon on the left is an option, to be consistent with similar lists:

Screenshot 2024-05-22 at 14 42 27
Screenshot 2024-05-22 at 14 42 00
Screenshot 2024-05-22 at 14 41 50
Screenshot 2024-05-22 at 14 45 22

File: packages/block-editor/src/components/block-bindings-toolbar-indicator/index.js (hunk 1 above)
Comment (richtabor) [reply packages/block-editor/src/components/block-bindings-toolbar-indicator/index.js:101]: I don't think we need a visual "Override" label.

**Comment by richtabor** [comment]
> I wonder if the connection's status message should be shown as a tooltip instead. Currently, it's a button that doesn't trigger any action, which can confuse users (including those using a screen reader).

It's already a button that does nothing, with a tooltip of the block name.

**Comment by richtabor** [comment]
Here’s the current state when selecting an overridable block while editing a page:

CleanShot 2024-05-31 at 11 20 41

**Comment by richtabor** [comment]
We don't need that button/application when editing the synced pattern entity directly. Results in a double-icon: 

CleanShot 2024-05-31 at 10 46 38

For this initial iteration, let's have this modification apply only in the context of overriding, rather than editing the synced pattern itself. I'm ok with a fast-follow bug fix for this as well.

**Review by richtabor — APPROVED** [review]
There are a lot of ideas on this, but the centered concept of using the block icon (purple) with a popover referencing details on what the block consists of is an improvement. 

Let's follow-up with a fix for the doubled icon when editing a pattern in focus mode (https://github.com/WordPress/gutenberg/pull/61560#issuecomment-2142432095), but this looks good to me.

**Comment by gziolo** [comment]
I run into an issue when testing Block Bindings in the site editor - homepage template. I know that establishing a post meta connection isn’t the best choice there but regardless, we need to prevent this type of accidental errors:

https://github.com/WordPress/gutenberg/assets/699132/96e0b59e-be15-4bd7-a28c-898dfea04ef5


It seems to be related to the changes introduced in this PR.

**Comment by SantosGuillamot** [comment]
It seems the issue is caused because at some point `block` [here](https://github.com/WordPress/gutenberg/pull/61560/files#diff-53c7becd18a8bab09d6e368f39a58eacb5ac5dcb25a80b2826308dedeaafc570R60) is undefined. So it isn't enough to use the optional operator, because we can't use `Object.values` in `undefined`. I guess it can be solved using something like this:

```js
Object.values( block?.attributes?.metadata?.bindings || {} )
```

**Comment by SantosGuillamot** [comment]
I started [this pull request](https://github.com/WordPress/gutenberg/pull/62268) with the potential fix. Although I must say I don't have the full context about that code.

**Pull request merged**